packages/dev/docs/src/docs.js

```diff
@@ -36,29 +37,56 @@ for (let link of links) {
         <Pressable>
           {/* eslint-disable jsx-a11y/click-events-have-key-events */}
           {/* eslint-disable jsx-a11y/anchor-is-valid */}
-          <a role="link" tabIndex={0} data-link={link.dataset.link} className={link.className} onClick={e => e.preventDefault()}>{link.textContent}</a>
+          <a role="button" tabIndex={0} aria-haspopup="dialog" data-link={link.dataset.link} className={link.className} onClick={e => e.preventDefault()}>{link.textContent}</a>
         </Pressable>
         <LinkPopover id={link.dataset.link} />
       </DialogTrigger>
     </ThemeProvider>
   , container);
 
-  link.parentNode.replaceChild(container, link);
+  link.parentNode.replaceChild(container.firstElementChild, link);
 }
 
 function LinkPopover({id}) {
   let ref = useRef();
   let [breadcrumbs, setBreadcrumbs] = useState([document.getElementById(id)]);
 
+  const onBlurCurrentBreadcrumb = useCallback((event) => {
+    event.target.removeEventListener('blur', onBlurCurrentBreadcrumb);
+    event.target.tabIndex = -1;
+  },
+  []);
+
   useEffect(() => {
+    let isCancelled = false; 
+    let timeoutId;
     let links = ref.current.querySelectorAll('[data-link]');
     for (let link of links) {
+      link.removeAttribute('aria-haspopup');
       link.addEventListener('click', (e) => {
         e.preventDefault();
         setBreadcrumbs([...breadcrumbs, document.getElementById(link.dataset.link)]);
       });
     }
-  }, [breadcrumbs]);
+
+    if (isCancelled) {
+      if (timeoutId) {
+        cancelAnimationFrame(timeoutId);
+      }
+      return;
+    }
+
+    timeoutId = requestAnimationFrame(() => {
+      const currentBreadcrumb = ref.current.closest(`.${docsStyle.popover}`).querySelector('[aria-current]');
+      if (currentBreadcrumb) {
+        currentBreadcrumb.tabIndex = 0;
+        currentBreadcrumb.addEventListener('blur', onBlurCurrentBreadcrumb);
+        focusWithoutScrolling(currentBreadcrumb); 
+      }
+    });
+
+    return () => isCancelled = true;
```

Comment:
the actual cancelling should take place in the useEffect cleanup, isCancelled will be a different instance in the next run of the useEffect, so I don't really see how this is currently doing anything

Reply:
Yeah, it's probably not doing anything, and I'm not sure that it's needed.

Reply:
Should be fixed now.